explorer/templates.go

```diff
@@ -259,5 +268,18 @@ func makeTemplateFuncMap(params *chaincfg.Params) template.FuncMap {
 		"theme": func() string {
 			return netTheme
 		},
+		"emptyArray": func(length int) []int {
+			if length < 0 {
+				length = 0
+			}
+			return make([]int, length)
+		},
+		"first30Txs": func(txs []TrimmedTxInfo) []TrimmedTxInfo {
+			first30 := make([]TrimmedTxInfo, 0, 30)
+			for i := 0; i < 30; i++ {
+				first30[i] = txs[i]
```

Comment:
Please use the built-in `copy` instead of a loop.  Also note that this panics if `txs` is shorter than 30 (or `N`).

Or even better, just create the new array with indexing: `front := txs[:N]` (when len(txs)>=N).

Reply:
Get comfy with all the concepts used in https://github.com/golang/go/wiki/SliceTricks